worldedit-core/src/main/java/com/fastasyncworldedit/core/function/mask/AngleMask.java

```diff
@@ -14,6 +14,7 @@ public class AngleMask extends AbstractExtentMask implements ResettableMask {
     protected static double ADJACENT_MOD = 0.5;
     protected static double DIAGONAL_MOD = 1 / Math.sqrt(8);
 
+    private final SolidBlockMask fastMask;
```

Comment:
`CachedMask` is somewhat expensive, I'm not exactly sure why but I assume due to the need of exception handling. Skipping it when we pass a CharFilterBlock as position seems to be always faster.

Reply:
Which exception handling exactly? It shouldn't be slow unless we're filling the stacktrace

Reply:
Just the presence of the exception region making optimizations harder. But that's just a wild guess, the profiler shows that 2/3 of the time in `CachedMask#test(Extent, BV3)` is actually spent in the method itself rather than in methods it is calling. But that might also be misattribution due to lacking debug info or something like that.

Also something I forgot to mention: The `CachedMask` is only really useful if `overlay` is true, otherwise positions won't be checked multiple times normally.

Reply:
> I don't think it should have any effect on performance though, unless you're testing the mask on a large range where the exception would actually be thrown? Afaik the try-catch basically means nothing until the exception walks back up the stack

I'm not sure what's going on exactly there, but I didn't want to do a deep-dive before merging your changes. I'll take a closer look once both changes are merged, that should give a clearer image.